table/get_context.cc

```diff
@@ -98,18 +100,27 @@ bool GetContext::SaveValue(const ParsedInternalKey& parsed_key,
       case kTypeValue:
         assert(state_ == kNotFound || state_ == kMerge);
         if (kNotFound == state_) {
-          state_ = kFound;
-          if (value_ != nullptr) {
-            value_->assign(value.data(), value.size());
+          if (range_del_agg_ != nullptr &&
```

Comment:
Same as previous comment. Could you take the range_del condition out of the switch block. Then switch the case between merge and non-merge cases.